src/http.cc

```diff
@@ -152,126 +152,155 @@ minio::error::Error minio::http::BaseUrl::BuildUrl(utils::Url &url,
   return error::SUCCESS;
 }
 
-size_t minio::http::Response::ReadStatusCode(char *buffer, size_t size,
```

Comment:
@balamurugana shouldn't we be using some standard HTTP clients here? instead of writing low-level logic ourselves?

Reply:
`libcurl` C++ wrappers like `curlpp`, `cpr` etc aleady provides this functionality. But it doesn't work for us in case of failed REST requests. Our requirement is, if REST request is succeeded, send body to passed callback function; else capture the body for getting error XML.  `libcurl` provides a `WRITEFUNCTON` callback for receiving response body. In the callback it is not possible to get HTTP status code to success/failed REST request. By this fact, there is no option other than receiving full response and get status code.

Reply:
Let's take it in but - but also lets discuss with @abperiasamy once - it is very surprising that C++ doesn't have a decent HTTP library.

Reply:
Any C++ wrapper of `libcurl` has this limitation. Another option is to use different kind of library. As `libcurl` is very stable,  solid library and feature rich, it is hard to say about other libraries. I am taking this PR and will move another library if needed as per our discussion.